From fefd6dcc8c5af4dd4e593a351b6b5cd9d0bde5be Mon Sep 17 00:00:00 2001
From: Tim Lunn <tim@feathertop.org>
Date: Wed, 20 Feb 2013 14:22:35 +1100
Subject: [PATCH] use MOZILLA_VERSION for JS_GetImplementationVersion()

---
 js/src/jsapi.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/jsapi.cpp b/js/src/jsapi.cpp
index 83ee6cd..b89911e 100644
--- a/js/src/jsapi.cpp
+++ b/js/src/jsapi.cpp
@@ -1409,7 +1409,7 @@ JS_SetJitHardening(JSRuntime *rt, JSBool enabled)
 JS_PUBLIC_API(const char *)
 JS_GetImplementationVersion(void)
 {
-    return "JavaScript-C 1.8.5+ 2011-04-16";
+    return "JavaScript-C " MOZILLA_VERSION;
 }
 
 JS_PUBLIC_API(void)
-- 
1.8.1.2



